pkg/executor/cluster.go

```diff
@@ -94,59 +42,18 @@ func RemoveFinalizer(
 	reqLogger logr.Logger,
 	cr *kdv1.KubeDirectorCluster,
 ) error {
-
-	if !removeFinalizer(cr) {
-		return nil
+	found := false
+	for i, f := range cr.Finalizers {
+		if f == finalizerID {
+			cr.Finalizers = append(cr.Finalizers[:i], cr.Finalizers[i+1:]...)
+			found = true
+			break
+		}
 	}
-	err := shared.Client().Update(context.TODO(), cr)
-	if err == nil {
+	if !found {
 		return nil
 	}
-	if !errors.IsConflict(err) {
-		shared.LogError(
-			reqLogger,
-			err,
-			cr,
-			shared.EventReasonCluster,
-			"failed to remove finalizer",
-		)
-		return err
-	}
-
-	// If there was a resourceVersion conflict then fetch a more
-	// recent version of the object and attempt to update that.
-	currentCluster := &kdv1.KubeDirectorCluster{}
-	err = shared.Client().Get(
-		context.TODO(),
-		types.NamespacedName{
-			Namespace: cr.Namespace,
-			Name:      cr.Name,
-		},
-		currentCluster,
-	)
-	if err != nil {
-		shared.LogError(
-			reqLogger,
-			err,
-			cr,
-			shared.EventReasonCluster,
-			"failed to retrieve cluster",
-		)
-		return err
-	}
-	if removeFinalizer(currentCluster) {
-		err = shared.Client().Update(context.TODO(), currentCluster)
-		if err != nil {
-			shared.LogError(
-				reqLogger,
-				err,
-				cr,
-				shared.EventReasonCluster,
-				"failed to remove finalizer",
-			)
-		}
-	}
-	return err
+	return shared.Client().Update(context.TODO(), cr)
```

Comment:
Should have the comment here about moving to PATCH when available.

Reply:
Done.